src/Symfony/Component/Security/Core/Tests/Dumper/MermaidDumperTest.php

```diff
@@ -0,0 +1,130 @@
+    }
+
+    public function testDumpWithDirection()
+    {
+        $hierarchy = [
+            'ROLE_ADMIN' => ['ROLE_USER'],
+        ];
+
+        $roleHierarchy = new RoleHierarchy($hierarchy);
+        $dumper = new MermaidDumper(MermaidDumper::DIRECTION_LEFT_TO_RIGHT);
+        $output = $dumper->dump($roleHierarchy);
+
+        $this->assertStringContainsString('graph LR', $output);
+    }
+
+    public function testDumpEmptyHierarchy()
+    {
+        $roleHierarchy = new RoleHierarchy([]);
+        $dumper = new MermaidDumper();
+        $output = $dumper->dump($roleHierarchy);
+
+        $this->assertStringContainsString('graph TB', $output);
+        $this->assertStringContainsString('classDef default fill:#e1f5fe;', $output);
+    }
+
+    public function testDumpComplexHierarchy()
+    {
+        $hierarchy = [
+            'ROLE_SUPER_ADMIN' => ['ROLE_ADMIN', 'ROLE_ALLOWED_TO_SWITCH'],
+            'ROLE_ADMIN' => ['ROLE_USER'],
+            'ROLE_MANAGER' => ['ROLE_USER'],
+            'ROLE_EDITOR' => ['ROLE_USER'],
+        ];
+
+        $roleHierarchy = new RoleHierarchy($hierarchy);
+        $dumper = new MermaidDumper();
+        $output = $dumper->dump($roleHierarchy);
+
+        // Check that all roles are present
+        $this->assertStringContainsString('ROLE_SUPER_ADMIN', $output);
+        $this->assertStringContainsString('ROLE_ADMIN', $output);
+        $this->assertStringContainsString('ROLE_MANAGER', $output);
+        $this->assertStringContainsString('ROLE_EDITOR', $output);
+        $this->assertStringContainsString('ROLE_USER', $output);
+        $this->assertStringContainsString('ROLE_ALLOWED_TO_SWITCH', $output);
+
+        // Check relationships
+        $this->assertStringContainsString('ROLE_SUPER_ADMIN --> ROLE_ADMIN', $output);
+        $this->assertStringContainsString('ROLE_SUPER_ADMIN --> ROLE_ALLOWED_TO_SWITCH', $output);
+        $this->assertStringContainsString('ROLE_ADMIN --> ROLE_USER', $output);
+        $this->assertStringContainsString('ROLE_MANAGER --> ROLE_USER', $output);
+        $this->assertStringContainsString('ROLE_EDITOR --> ROLE_USER', $output);
+    }
+
+    public function testInvalidDirection()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Direction "INVALID" is not valid');
+
+        new MermaidDumper('INVALID');
+    }
+
+    public function testValidDirections()
+    {
+        $validDirections = [
+            MermaidDumper::DIRECTION_TOP_TO_BOTTOM,
+            MermaidDumper::DIRECTION_TOP_DOWN,
+            MermaidDumper::DIRECTION_BOTTOM_TO_TOP,
+            MermaidDumper::DIRECTION_RIGHT_TO_LEFT,
+            MermaidDumper::DIRECTION_LEFT_TO_RIGHT,
+        ];
+
+        foreach ($validDirections as $direction) {
+            $dumper = new MermaidDumper($direction);
+            $this->assertInstanceOf(MermaidDumper::class, $dumper);
+        }
```

Comment:
There's no point to instantiate an object then assert its type. If you want to test valid directions, I'd suggest instead to use `@testWith`, instantiate the dumper and `$this->expectNoAssertion()`. Also, `testWith` will force you to use the raw values of constants, which is nice because tests will break (and warn) if someone updates one of the values